### PR TITLE
KIN-242 Validation for Ladas

### PR DESCRIPTION
**Pull request by FernandoJoachin**
[KIN-242](https://loopcrack.atlassian.net/browse/KIN-242) Medium [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

The validation of the telephone field was changed so that it validates Ladas.

### Problem

The form does not accept and valid Ladas.

### Solution

Put a regular expression to accept and validate Ladas.

### Requirements checklist

Include a list of requirements necessary to consider this pull request complete

- [x] The telephone field must allow placing and validating Lada.

### Testing
it´s neccesary to install the new dependency using  `npm install`

- [x] Select Mexico and enter a phone number less than 10 digits. This should show a validation alert. For example 222 123
- [x] Select Mexico and enter a phone number greater than 10 digits. This should show a validation alert. For example 222 123 456799
- [x] Select Mexico and enter a valid phone number. The program should display the success alert. For example 5523457812
- [x] Select another country like Andorra and write a 6-digit number and pass the validation since in that country the telephone number is 6 digits. For example: 334 456
- [x] Enter a valid phone number in any type of format and the success alert should be displayed. For example (552)-345-7812
- [x] Check that valid phone numbers have been saved in the database and sent to emails with the E.164 format (+CCNNNNNNNNNN)

[KIN-242]: https://loopcrack.atlassian.net/browse/KIN-242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/Validation/SupportEmailValidation.php [hunk 1]
```diff
@@ -21,8 +21,7 @@ class SupportEmailValidation
         ],
         "support-phone" => [
             "required" => "El teléfono es obligatorio",
-            "numeric" => "El campo teléfono solo debe contener carácteres numéricos",
-            "exact_length" => "El teléfono debe tener 10 dígitos"
+            "regex_match" => "El campo teléfono debe contener 10 dígitos numéricos y puede incluir una lada opcional"
```
Comment (CarlosMay7) [inline app/Validation/SupportEmailValidation.php:24]:
```suggestion
            "regex_match" => "El campo teléfono debe contener 10 dígitos y puede incluir una lada opcional"
```
Todo muy bien, solo podrías cambiar esta parte ya que es algo reduntante, pues un dígito ya es numérico

**Review by CarlosMay7 — APPROVED** [review]

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]
1. Guardar en Formato E.164:
Es aconsejable guardar los números en formato E.164: +CC NNNNNNNNNN, donde +CC es el código del país y NNNNNNNNNN es el número local.
El formato E.164 es estándar y es reconocido internacionalmente, lo que facilita el manejo de números internacionales.
2. No Guardar Formateo:
Evitar guardar espacios, guiones, paréntesis u otros caracteres de formateo.
El formateo se puede realizar en la aplicación al mostrar el número al usuario.
3. Lidiar con Números Locales:
Si recibes números locales sin código de país, podrías considerar agregar el código del país por defecto.
Asegurarte de que los usuarios sepan que deben ingresar el código de país si están proporcionando un número internacional.

**Review by LUGAMAFE — APPROVED** [review]
@FernandoJoachin hice un commit haciendo unas correcciones a tu codigo en vez de dejarte comentarios pero en resumen añadi [Async Validators](https://welisonmenezes.github.io/vanilla-validator/documentation/async-validators.html) que es parte de la documentacion del plugin que añadio @Oswaldo-Chan para mejorar la logica de lo que hiciste

**Comment by LUGAMAFE** [comment]
@FernandoJoachin porfavor sube un video mostrando la demo de este PR hiciste no llenaste correctamente el template del PR ojo!

**Review by Mauriciocr207 — APPROVED** [review]

**Review by Oswaldo-Chan — APPROVED** [review]

File: src/public/js/support.js [hunk 2]
```diff
@@ -1,129 +1,158 @@
...
-        setContainerVisibility(containerErrors, shouldDisplayError, cls, message);
+      },
+    },
+    onContainerSuccess: function () {
+      currentStep++;
+      if (currentStep == 4) {
+        const phoneNumber = phoneInput.getNumber(
+          intlTelInputUtils.numberFormat.E164
+        );
+        phoneInputField.value = phoneNumber;
```
Comment (Mauriciocr207) [inline src/public/js/support.js:77]: I think, there´s a problem on the line 77 when it set the `phoneNumber` into `phoneInputField`. Here what I see:
First, if you try to pass to the next section of the form introducing wrong data into name and email inputs on first step of form, you clearly can not pass to the next step. 
image

But, if you let the wrong data in the phone input, you **can pass** to next step
image

image

On the other hand, if you fill all the inputs on form, but you let the input number something like this: `123asdasdasd`, you receive a message saying your number is incorrect, but like I said before, you can pass the next step
image
and you continue filling all other inputs:
image
image

you recieve a message saying your email was sended
image

And if you see the email, looks like if you have been sended a number, but you put `123asdasdasd` into your input number
image

**Review by Mauriciocr207 — CHANGES_REQUESTED** [review]

File: src/public/js/support.js (hunk 2 above)
Comment (Mauriciocr207) [reply src/public/js/support.js:77]: I forgot mentioned @LUGAMAFE @FernandoJoachin

Comment (LUGAMAFE) [reply src/public/js/support.js:79]: Ready @Mauriciocr207

**Review by Mauriciocr207 — APPROVED** [review]

**Review by CarlosJCD — APPROVED** [review]

**Pull request merged**